sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1261,6 +1397,138 @@ no queue state is changed. This design is because the queues are already in
+----
+/* Basic usage example using the Record&Reply API. */
+
+// Create the graph `A`.
+sycl::queue queueA;
+sycl_ext::command_graph graphA(queueA.get_context(), queueA.get_device());
+
+// Start recording the queue.
+graphA.begin_recording(queueA);
+
+// Submit kernel on the recorded queue.
+// EventA is a graph-limited event that can only be used for managing
+// graph internal dependencies.
+EventA = queueA.submit([&](handler& cgh) {...});
+
+// Stop recording the queue.
+graphA.end_recording();
+
+event externalEventA;
+queueA.ext_oneapi_external_event(EventA, externalEventA, flase /* non-updatable */);
+
+// Create another graph `B`
+sycl::queue queueB;
+sycl_ext::command_graph graphB(queueB.get_context(), queueB.get_device());
+
+// Start recording the queue.
+graphB.begin_recording(queueB);
+
+// Submit kernel on the recorded queue.
+queueB.submit([&](handler& cgh) {...});
+
+queueB.ext_oneapi_submit_barrier({externalEventA});
+
+// Stop recording the queue.
+graphB.end_recording();
+
+// Finalize the modifiable graphs to create an executable graphs that can be
+// submitted for execution.
+auto execGraphA = graphA.finalize();
+auto execGraphB = graphB.finalize();
+
+// Execute graph
+for(int i = 0 ; i < Iterations, i++) {
+  queueA.submit([&](handler& cgh) {
+    cgh.ext_oneapi_graph(execGraphA);
+  })
+
+  queueB.submit([&](handler& cgh) {
+    cgh.ext_oneapi_graph(execGraphB);
+  })
+}
+----
+
+[source,c++]
+----
+/* Example: Event update. */
+
+// Create the graph `A`.
+sycl::queue queueA;
+sycl_ext::command_graph graphA(queueA.get_context(), queueA.get_device());
+
+// Start recording the queue.
+graphA.begin_recording(queueA);
+
+// Submit kernel on the recorded queue.
+// EventA is a graph-limited event that can only be used for managing
+// graph internal dependencies.
+EventA = queueA.submit([&](handler& cgh) {...});
+
+// Stop recording the queue.
+graphA.end_recording();
+
+event externalEventA;
+auto nodeSignalExternalEvent = graphA.make_external_event(EventA, 
+                                                          externalEventA, 
+                                                          true /* updatable */);
+
+// Create another graph `B`
+sycl::queue queueB;
+sycl_ext::command_graph graphB(queueB.get_context(), queueB.get_device());
+
+// Submit kernel on the recorded queue.
+graphB.add([&](handler& cgh) {...});
+
+auto nodeWaitExternalEvent = graphB.add_barrier({externalEventA});
+
+// Stop recording the queue.
+graphB.end_recording();
+
+// Finalize the modifiable graphs to create an executable graphs that can be
+// submitted for execution.
+auto execGraphA = graphA.finalize();
+auto execGraphB = graphB.finalize();
+
+// Execute graph
+// first execution
+queueA.submit([&](handler& cgh) {
+  cgh.ext_oneapi_graph(execGraphA);
+})
+queueB.submit([&](handler& cgh) {
+  cgh.ext_oneapi_graph(execGraphB);
+})
+
+event newExternalEventA;
+execGraphA.update_external_event(nodeSignalExternalEvent, newExternalEventA);
```

Comment:
If events are reset when the graph is re-executed, I'm not sure in which use cases updating the signal command might be useful. But the CUDA graph API offers this option for users [here](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__GRAPH.html#group__CUDART__GRAPH_1g79d6d9d5868d8f3ac1902fceee36ffeb). So I assume it might be useful in some workloads.

Reply:
I would expect that it would be useful to update an external event that is a dependency for a graph, though I'm not sure whether it would be useful to update an event produced by a graph, though I'm not very experienced with CUDA graphs. As the same API would be used for both cases, I don't see any harm in allowing it, though perhaps we should explore CUDA use cases for this capability further to get a better idea of how this would be used in SYCL.

Reply:
Finding actual real world uses (or even examples/documentation) of CUDA functionality in this area is quite challenging, so maybe we should err on the side of caution and leave out updating these nodes for now. It's easy enough to add if someone comes to us with a use case.